universal-application-tool-0.0.1/app/services/export/CsvExporter.java

```diff
@@ -50,24 +57,30 @@ private void writeHeadersOnFirstExport(CSVPrinter printer) throws IOException {
    * the writer between calls. Since it is intended for many applications, this function is intended
    * to be called several times.
    */
-  public void export(Application application, Writer writer) throws IOException {
+  public void export(
+      Application application, ReadOnlyApplicantProgramService roApplicantService, Writer writer)
+      throws IOException {
     CSVPrinter printer = new CSVPrinter(writer, CSVFormat.DEFAULT.withFirstRecordAsHeader());
 
     this.writeHeadersOnFirstExport(printer);
 
+    ImmutableMap<String, AnswerData> answerMap = roApplicantService.getSummaryDataMap();
```

Comment:
Why can't this be done for the demographic export again?

Reply:
It probably can. The key, however, is highly dependent on the program (basically it's block id + question index in the block). So for each question I will have to find the programs they are in, and generate the correct key based on the application. I feel that's too much of a trouble than doing this, but maybe it's not that hard.

Reply:
I can actually just make all answers in the application a map and select by Path.
hmm I guess it's doable. I will give it a try.

Reply:
If you can that would make me happy because it makes some of my job a lot easier, ha.  Thanks for giving it a shot!

Reply:
Done.